jax/interpreters/partial_eval.py

```diff
@@ -150,16 +150,18 @@ def default_process_primitive(self, primitive, tracers, params):
     tracers = map(self.instantiate_const, tracers)
     avals = [t.aval for t in tracers]
     out_aval = primitive.abstract_eval(*avals, **params)
-    if primitive.multiple_results:
+    
+    def create_output_tracers(out_avals):
```

Comment:
How about we tweak it so that instead of defining `mak_output_tracer(out_avals : List[AbstractValue])` we define a `make_output_tracer(out_aval: AbstractValue)` and then write

```python
if primitive.multiple_results:
  return map(make_output_tracer, out_aval)
else:
  return make_output_tracer(out_aval)
```

WDYT? Then we can avoid the `f([x])[0]` pattern at the bottom.

Reply:
I don't think that's possible, as all output tracers must share the same `eqn`.